xarray/tests/test_variable.py

```diff
@@ -709,6 +750,141 @@ def test_items(self):
+        self.assertTrue(v_new.dims == ('x', 'y'))
+        self.assertArrayEqual(v_new, v.load()._data[0][1])
+
+        ind = Variable(['a'], [True, False])
+        v_new = v[dict(y=ind)]
+        self.assertTrue(v_new.dims == ('x', 'a'))
+        self.assertArrayEqual(v_new, v.load()._data[:, 0:1])
+
+    def test_getitem_fancy(self):
+        # Note  This fancy getitem is not supported by dask-based Variable.
+        v = self.cls(['x', 'y'], [[0, 1, 2], [3, 4, 5]])
+
+        ind = Variable(['a', 'b'], [[0, 1, 1], [1, 1, 0]])
+        v_new = v[ind]
+        self.assertTrue(v_new.dims == ('a', 'b', 'y'))
+        self.assertArrayEqual(v_new, v.load()._data[([0, 1, 1], [1, 1, 0]), :])
+
+        ind = Variable(['a', 'b'], [[0, 1, 2], [2, 1, 0]])
+        v_new = v[dict(y=ind)]
+        self.assertTrue(v_new.dims == ('x', 'a', 'b'))
+        self.assertArrayEqual(v_new, v.load()._data[:, ([0, 1, 2], [2, 1, 0])])
+
+        ind = Variable(['a', 'b'], [[0, 0], [1, 1]])
+        v_new = v[dict(x=[1, 0], y=ind)]
+        self.assertTrue(v_new.dims == ('x', 'a', 'b'))
+        self.assertArrayEqual(v_new, v.load()._data[[1, 0]][:, ind])
+
+        # with integer
+        ind = Variable(['a', 'b'], [[0, 0], [1, 1]])
+        v_new = v[dict(x=0, y=ind)]
+        self.assertTrue(v_new.dims == ('a', 'b'))
+        self.assertArrayEqual(v_new[0], v.load()._data[0][[0, 0]])
+        self.assertArrayEqual(v_new[1], v.load()._data[0][[1, 1]])
+
+        # with slice
+        ind = Variable(['a', 'b'], [[0, 0], [1, 1]])
+        v_new = v[dict(x=slice(None), y=ind)]
+        self.assertTrue(v_new.dims == ('x', 'a', 'b'))
+        self.assertArrayEqual(v_new, v.load()._data[:, [[0, 0], [1, 1]]])
+
+        ind = Variable(['a', 'b'], [[0, 0], [1, 1]])
+        v_new = v[dict(x=ind, y=slice(None))]
+        self.assertTrue(v_new.dims == ('a', 'b', 'y'))
+        self.assertArrayEqual(v_new, v.load()._data[[[0, 0], [1, 1]], :])
+
+        ind = Variable(['a', 'b'], [[0, 0], [1, 1]])
+        v_new = v[dict(x=ind, y=slice(None, 1))]
+        self.assertTrue(v_new.dims == ('a', 'b', 'y'))
+        self.assertArrayEqual(v_new,
+                              v.load()._data[[[0, 0], [1, 1]], slice(None, 1)])
+
+    def test_getitem_error(self):
+        v = self.cls(['x', 'y'], [[0, 1, 2], [3, 4, 5]])
+
+        with self.assertRaisesRegexp(IndexError, "Unlabelled multi-"):
+            v[[[0, 1], [1, 2]]]
+
+        with self.assertRaisesRegexp(ValueError, "operands cannot be "):
+            ind_x = Variable(['a', 'b'], [[0, 0], [1, 1]])
+            ind_y = Variable(['a'], [0])
+            v[(ind_x, ind_y)]
+
+        with self.assertRaisesRegexp(IndexError, "2-dimensional boolean"):
+            ind = Variable(['a', 'b'], [[True, False], [False, True]])
+            v[dict(x=ind)]
+
+    def test_setitem(self):
+        v = self.cls(['x', 'y'], [[0, 3, 2], [3, 4, 5]])
+        v[0, 1] = 1
+        self.assertTrue(v[0, 1] == 1)
```

Comment:
No need to change these ex-post, but for tests in the future: we are moving tests to be pytesty - so `assert v[0, 1] == 1` is fine here, and above you can use `assert_equal` from `xr.testing` rather than `self.assertArrayEqual`, for example.
Thanks!

You may also be able to do some of these using parametrization